pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1395,6 +1418,43 @@ func (r *ReconcileClusterDeployment) removeClusterDeploymentFinalizer(cd *hivev1
 	return nil
 }
 
+func (r *ReconcileClusterDeployment) releaseClusterDeploymentCustomization(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	customizationRef := cd.Spec.ClusterPoolRef.ClusterDeploymentCustomizationRef
+	namespace := types.NamespacedName{Namespace: cd.Namespace, Name: customizationRef.Name}
+	cdc := &hivev1.ClusterDeploymentCustomization{}
+	err := r.Get(context.TODO(), namespace, cdc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			cdLog.Infof("customization not found: %s, nothing to release", customizationRef.Name)
+			controllerutils.DeleteFinalizer(cd, hivev1.FinalizerCustomizationRelease)
+			return nil
+		}
+		log.WithError(err).Error("error reading customization")
+		return err
+	}
+
+	cdc.Status.ClusterDeploymentRef = nil
+	cdc.Status.Conditions = controllerutils.SetClusterDeploymentCustomizationCondition(
+		cdc.Status.Conditions,
+		hivev1.ClusterDeploymentCustomizationAvailableCondition,
+		corev1.ConditionTrue,
+		"ClusterDeploymentCustomizationReleased",
+		"Cluster Deployment Customization was released",
```

Comment:
I would go with the message that says customization is available instead of mentioning it was released